### ISSUE TITLE
TCastleLabel – intermittent updating of position, scale

### ISSUE THREAD

**Issue by e1e5en-gd** [castle-engine/castle-engine#494]
When changing the `TCastleLabel` in the `Update` method, changes occur intermittently.
For example, zoom changes using `FontScale`.
https://github.com/castle-engine/castle-engine/assets/7114353/44d40333-33ad-47c4-a344-4cbaaf9b7d61

When changing the position to a "small" distance, this is clearly visible.
https://github.com/castle-engine/castle-engine/assets/7114353/7ca473fa-0164-4426-ad01-162e7e1ef4af

If you change the positions for a "long" distance, it is not so noticeable.
https://github.com/castle-engine/castle-engine/assets/7114353/02593fac-99ab-4e9e-a8f7-5c67820a80fc

An example of a project. All modifiers in the `Update` method: 
[bubble-0.1-src.zip](https://github.com/castle-engine/castle-engine/files/11518152/bubble-0.1-src.zip)

**Comment by eugeneloza** [comment]
@michaliskambi I've found the culprit! So, as we talked about this on Wednesday, indeed there is some rounding happening in `TFont.Print`:

https://github.com/castle-engine/castle-engine/blob/master/src/fonts/castlefonts_font.inc#L332

However, now I'm in doubt. As rounding here indeed makes perfect sense. As it "hurts" only animated moving labels and benefits font sharpness in all other cases. I guess both the solutions we've discussed : Distance Field Font and Image Label would help greatly with this, but in the current "simple" paradigm it may do more harm than good. WDYT?

**Comment by eugeneloza** [comment]
No, I didn't test it, but looks way too obvious. Of course I can make a test.

**Comment by eugeneloza** [comment]
Hmmm, fixes the issue for me.

After fix:

https://github.com/castle-engine/castle-engine/assets/10726162/95eb7562-5488-474a-8c96-5841b9fad4f0

Before fix:

https://github.com/castle-engine/castle-engine/assets/10726162/47267cd9-825e-483b-9e29-593608387aa4
